sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -375,63 +407,76 @@ export namespace ConnectionContext {
         await connectionContext.managementClients[entityPath].close();
       }
 
-      // Calling onDetached on sender
-      if (!state.wasConnectionCloseCalled && state.numSenders) {
-        // We don't do recovery for the sender:
-        //   Because we don't want to keep the sender active all the time
-        //   and the "next" send call would bear the burden of creating the link.
-        // Call onDetached() on sender so that it can gracefully shutdown
-        //   by cleaning up the timers and closing the links.
-        // We don't call onDetached for sender after `refreshConnection()`
-        //   because any new send calls that potentially initialize links would also get affected if called later.
-        logger.verbose(
-          `[${connectionContext.connection.id}] connection.close() was not called from the sdk and there were ${state.numSenders} ` +
-            `senders. We should not reconnect.`
-        );
-        const detachCalls: Promise<void>[] = [];
-        for (const senderName of Object.keys(connectionContext.senders)) {
-          const sender = connectionContext.senders[senderName];
-          if (sender) {
-            logger.verbose(
-              "[%s] calling detached on sender '%s'.",
-              connectionContext.connection.id,
-              sender.name
-            );
-            detachCalls.push(
-              sender.onDetached().catch((err) => {
-                logger.logError(
-                  err,
-                  "[%s] An error occurred while calling onDetached() the sender '%s'",
-                  connectionContext.connection.id,
-                  sender.name
-                );
-              })
-            );
+      if (state.wasConnectionCloseCalled) {
+        // Do Nothing
+      } else {
```

Comment:
Just curious... what was the reason to pull the check on wasConnectionCloseCalled out?

Reply:
`if (!state.wasConnectionCloseCalled)` was being done for senders, non-session batching receivers, and sessionful baching receivers separately.
Pulling it out helps us with moving the common code together.

(@richardpark-msft suggested moving the check in one of the comments so that it makes it easier to follow through the code.)

Reply:
Early exit would also have been a good option here but I like your way too.